### PR TITLE
AB#77434 - Improve duplication of application

### PR DESCRIPTION
**Pull request by guisoares1**
# Description

**Situations found in tests and fixed:**

- [x] Page and subpage icons (internal page tabs) are not being copied. [Commit link.](https://github.com/ReliefApplications/ems-backend/commit/30a5a030e46d420b66a2aa90e165366a00c8af05)
- [x] Page and steps sequence is being changed. [Commit link.](https://github.com/ReliefApplications/ems-backend/commit/30a5a030e46d420b66a2aa90e165366a00c8af05)
- [x] When duplicating an application, deleted pages were restored in the new application. [Commit link.](https://github.com/ReliefApplications/ems-backend/commit/a2bb26cdb339ac6296bbeea22606eb8a662d5d27)
- [x] When an application is duplicated, the access settings are not being copied. [Commit link.](https://github.com/ReliefApplications/ems-backend/pull/782/commits/8735fda72109523100da0c591bf17a1afeba4a8c)
- [x] When the application is duplicated it is not updating the list (interface) as happens when you delete an application. [Back-end branch](https://github.com/ReliefApplications/ems-frontend/pull/1987)
## Useful links

- [Link to ticket.](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/77434)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (refactor or addition to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [x] An application with problematic characteristics was created and behavior was observed in the new application.

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-backend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by matheus-relief** [comment]
@guisoares1 @AntoineRelief Improved a couple things:
* Copy context and contextualized pages from source app
* Copy resources (including field level) permissions for the application roles created

**Review by AntoineRelief — CHANGES_REQUESTED** [review]
@matheus-relief 
almost all good !
but there are some elements that are not yet correctly duplicated

Indeed, so far I can see that distribution lists & notification templates are not duplicated
In addition, if you have a grid widget using one of the notification template / distribution list, when you duplicate the whole application you should be able to retrieve them
perhaps a regex in the structure of the widget could replace the old ids with the new ones

that will also affect some other widgets I believe
for example, text & summary card widgets may contain a link to another page of the application. In that case, it would be good to replace the id in the link, with the duplicated one ( a simple regex should do it )
in summary card / grid widgets, we also have the possibility to redirect to another page ( see new settings in enabled actions about that ). We would have the same issue

**Comment by matheus-relief** [comment]
@AntoineRelief Did all the changes requested, also am duplicating the Position atrributes category, don't know if I should. There is still the notification templates and subscriptions missing, but I remember you saying we weren't using those? Let me know if I should add those as well

**Comment by AntoineRelief** [comment]
> @AntoineRelief Did all the changes requested. Also am duplicating the Position attributes category, don't know if I should. There is still the notification templates and subscriptions missing, but I remember you saying we weren't using those? Let me know if I should add them as well

please add everything @matheus-relief 
even if not used

**Comment by matheus-relief** [comment]
@AntoineRelief I think it's all good now!
